### PR TITLE
Chore: Move login attempt methods to separate service

### PR DESCRIPTION
**Pull request by zserge**
Related to #46522

### CODE REVIEW

File: pkg/services/cleanup/cleanup.go [hunk 1]
```diff
@@ -184,7 +188,7 @@ func (srv *CleanUpService) deleteOldLoginAttempts(ctx context.Context) {
 	cmd := models.DeleteOldLoginAttemptsCommand{
 		OlderThan: time.Now().Add(time.Minute * -10),
 	}
```
Comment (sakjur) [inline pkg/services/cleanup/cleanup.go:190]: QQ: This kind of cleanup job should eventually move over to the service, right?

Do we want a centralized cleanup service that cleanup jobs register with or do we want to register a whole lot of different but similar cleanup jobs as background jobs? I think having a single service similar to usage stats would make sense?

File: pkg/services/loginattempt/loginattemptimpl/store_test.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/loginattempt"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/stretchr/testify/require"
+)
+
+func mockTime(mock time.Time) time.Time {
+	getTimeNow = func() time.Time { return mock }
+	return mock
+}
```
Comment (sakjur) [inline pkg/services/loginattempt/loginattemptimpl/store_test.go:17]: We can turn this dog override to a cat override (…no?) by having it cover its tracks. WDYT?

```suggestion
func mockTime(t testing.TB, mock time.Time) time.Time {
	oldTimeNow = getTimeNow
	t.Cleanup(func() {
		getTimeNow = oldTimeNow
	})
	getTimeNow = func() time.Time { return mock }
	return mock
}
```

**Review by sakjur — APPROVED** [review]
✅

File: pkg/services/loginattempt/loginattemptimpl/store_test.go (hunk 2 above)
Comment (zserge) [reply pkg/services/loginattempt/loginattemptimpl/store_test.go]: I actually decided to implement `now()` as a struct field and not a global mock. Easier to fake, no need to clean up later.

Comment (sakjur) [reply pkg/services/loginattempt/loginattemptimpl/store_test.go]: Ready for parallelism 💯

**Pull request merged**